test/qa/test_dns.py

```diff
@@ -85,10 +87,44 @@ def test_custom_dns_connect(tech, proto, obfuscated, nameserver):
     assert dns.is_unset()
 
 
+@pytest.mark.parametrize("nameserver", lib.DNS)
+@pytest.mark.parametrize("tech,proto,obfuscated", lib.TECHNOLOGIES)
+@pytest.mark.flaky(reruns=2, reruns_delay=90)
+@timeout_decorator.timeout(40)
+def test_custom_dns_off_connect(tech, proto, obfuscated, nameserver):
+    lib.set_technology_and_protocol(tech, proto, obfuscated)
+
+    sh.nordvpn.set.dns(nameserver)
+    assert dns.dns_visible_in_settings(nameserver)
+    assert dns.is_unset()
+
+    sh.nordvpn.set.dns("off")
+    assert dns.dns_visible_in_settings("disabled")
+    assert dns.is_unset()
+
+    output = sh.nordvpn.connect()
+    print(output)
+    assert lib.is_connect_successful(output)
+
+    assert not dns.dig(nameserver)
+    assert dns.is_unset()
+
+    with lib.ErrorDefer(sh.nordvpn.disconnect):
+        assert network.is_connected()
```

Comment:
Is this check really necessary? I understand checking this in connect tests, but I think this is an overkill in DNS tests. I don't think everything should be checked every time, DNS tests should check DNS stuff. But maybe I'm wrong, this is just my opinion.